server/Dockerfile

```diff
@@ -1,132 +1,88 @@
 ARG DOCKER_HUB_PROXY=""
 
-FROM "${DOCKER_HUB_PROXY}debian:bullseye-slim" as composer-build
+# PHP7.4 explicitly because of ssdeep being incompatible with PHP 8.2
+FROM "${DOCKER_HUB_PROXY}ubuntu:22.04" as composer-build
+
     ENV DEBIAN_FRONTEND noninteractive
     ENV COMPOSER_ALLOW_SUPERUSER 1
     ARG MISP_TAG
     ARG MISP_COMMIT
+    RUN apt-get update && apt-get install -y --no-install-recommends gpg-agent software-properties-common && add-apt-repository -y ppa:ondrej/php \
+        && apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/*
     RUN apt-get update; apt-get install -y --no-install-recommends \
         ca-certificates \
-        php \
-        php-apcu \
-        php-curl \
-        php-xml \
-        php-intl \
-        php-bcmath \
-        php-mbstring \
-        php-mysql \
-        php-redis \
-        php-gd \
-        php-fpm \
-        php-zip \
+        php7.4 \
+        php7.4-apcu \
+        php7.4-curl \
+        php7.4-xml \
+        php7.4-intl \
+        php7.4-bcmath \
+        php7.4-mbstring \
+        php7.4-mysql \
+        php7.4-redis \
+        php7.4-gd \
+        php7.4-fpm \
+        php7.4-zip \
+        unzip \
         && apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/*
     
     WORKDIR /tmp
     ADD https://raw.githubusercontent.com/MISP/MISP/${MISP_COMMIT:-${MISP_TAG}}/app/composer.json /tmp
     COPY --from=composer:latest /usr/bin/composer /usr/bin/composer
     RUN composer config --no-interaction allow-plugins.composer/installers true
     RUN composer install
-    RUN composer require --with-all-dependencies \
+    RUN composer require --with-all-dependencies --no-interaction \
             supervisorphp/supervisor:^4.0 \
             guzzlehttp/guzzle \
             lstrojny/fxmlrpc \
             php-http/message \
-            php-http/message-factory
-    RUN composer require --with-all-dependencies \
+            php-http/message-factory \
```

Comment:
It wasn't immediately obvious to me why these were two separate layers, so I combined them

Reply:
Ah  yes, that's more obvious to me after you mentioned trying to follow INSTALL as closely as possible 👍 

Perhaps an inline comment rather than a layer?

Reply:
Added an inline comment for it